### PR TITLE
[Lock] fix derivating semaphore from key

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | -
| License       | MIT
| Doc PR        | -

A crc32 is a 32bits number.
This new code uses a 64bits number on x64.
This reduces the risk of collision (that could cause a deadlock) by a huge amount.

### CODE REVIEW

File: src/Symfony/Component/Lock/Store/SemaphoreStore.php [hunk 1]
```diff
@@ -63,7 +63,7 @@ private function lock(Key $key, bool $blocking)
             return;
         }
 
-        $keyId = crc32($key);
+        $keyId = unpack('i', md5($key, true))[1];
```
Comment (jderusse) [inline src/Symfony/Component/Lock/Store/SemaphoreStore.php:66]: What is the purpose of this change?

Don't we have the same code in Lock components?

Comment (nicolas-grekas) [reply src/Symfony/Component/Lock/Store/SemaphoreStore.php:66]: A crc32 is a 32bits number.
This new code uses a 64bits number on x64.
This reduces the risk of collision (that could cause a deadlock) by a huge amount.

Comment (nicolas-grekas) [reply src/Symfony/Component/Lock/Store/SemaphoreStore.php:66]: And this *is* the Lock component :)

**Review by jderusse — APPROVED** [review]

File: src/Symfony/Component/Lock/Store/SemaphoreStore.php (hunk 1 above)
Comment (stof) [reply src/Symfony/Component/Lock/Store/SemaphoreStore.php:66]: this could be mentioned in the commit message, to give context to the change in the git history
